src/main/java/pl/zalwi/TransactionController.java

```diff
@@ -0,0 +1,172 @@
+package pl.zalwi;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import pl.zalwi.data.Transaction;
+import pl.zalwi.data.TransactionType;
+import pl.zalwi.data.TransactionsRepository;
+import pl.zalwi.logic.TransactionDao;
+
+import java.math.BigDecimal;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Controller
+public class TransactionController {
+
+    private TransactionsRepository transactionsRepository;
+    private TransactionDao transactionDao;
+
+    @Autowired
+    public TransactionController(TransactionsRepository transactionsRepository, TransactionDao transactionDao) {
+        this.transactionsRepository = transactionsRepository;
+        this.transactionDao = transactionDao;
+    }
+
+    @GetMapping("/")
+    public String main(Model model) {
+        model.addAttribute("expense", TransactionType.EXPENSE);
+        model.addAttribute("income", TransactionType.INCOME);
+        return "home";
+    }
+
+    @GetMapping("/list")
+    public String listTransactions(@RequestParam(required = false, name = "type") Optional<TransactionType> optionalTransactionType, Model model) {
+
+        if(optionalTransactionType.isEmpty()){
+            model.addAttribute("listDescription", "wszystkie");
+            Optional<List<Transaction>> optionalTransactions = transactionDao.readList(null);
+            if(optionalTransactions.isPresent()){
+                model.addAttribute("items", optionalTransactions.get());
+            }else{
+                model.addAttribute("items", null);
```

Comment:
Tak też nie rób - jak `null` jest dla Ciebie poprawną wartością w logice biznesowej to najprawdopodobniej robisz coś nie tak.